src/nccl_ofi_rdma.c

```diff
@@ -1502,6 +1506,21 @@ static inline int process_err_completion(nccl_net_ofi_rdma_device_t *device,
 		goto exit;
 	}
 
+	if (err_entry.err == FI_ECANCELED) {
+		/* Closing an EP with posted receives can generate cancellation events
+		   for the posted receives */
+		req = err_entry.op_context;
+		if (req == NULL || req->type != NCCL_OFI_RDMA_BOUNCE) {
```

Comment:
If we are expecting a cancelation completion and squashing it, we should just explicitly cancel it ourselves before closing the EP in my opinion, so any other error completion indicates a problem elsewhere. But this works too. Hard to tell from this commit's diff, but to make sure, we already free up state for these requests right?

Reply:
> If we are expecting a cancelation completion and squashing it, we should just explicitly cancel it ourselves before closing the EP in my opinion, so any other error completion indicates a problem elsewhere.

Yeah, I thought about that originally and discussed with Brian. We figured that closing the endpoint basically does the same thing, and fetching and cancelling the individual requests would be a decent amount of extra work.

>  Hard to tell from this commit's diff, but to make sure, we already free up state for these requests right?

Yes. Thanks for pointing this out: I think this is actually a race condition. We've already freed the bounce buffer requests at this point ([link](https://github.com/rauteric/aws-ofi-nccl/blob/3291241d091848ff07b18b8de79815f8d43844b6/src/nccl_ofi_rdma.c#L5528)), so we shouldn't be accessing the request here.

Instead, here in the cancel handling code, we need to just check that the event is a receive op, since we only have one type of receive op in the plugin.

Reply:
> Yeah, I thought about that originally and discussed with Brian. We figured that closing the endpoint basically does the same thing, and fetching and cancelling the individual requests would be a decent amount of extra work.

Actually, we decided to go this route and explicitly cancel each recv req.